core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -107,13 +107,15 @@ public Schema record(Schema record, List<String> names, Iterable<Schema.Field> s
             field.name());
         // Create a field that will be defaulted to null. We assign a unique suffix to the field
         // to make sure that even if records in the file have the field it is not projected.
+        String fieldName = AvroSchemaUtil.makeCompatibleName(field.name());
```

Comment:
this is already being done in L98, can we maybe do `String avroFieldName = AvroSchemaUtil.makeCompatibleName(field.name());` in L97 and use `avroFieldName` in L113?

Reply:
Updated